docs/csharp/how-to/compare-strings.md

```diff
@@ -0,0 +1,86 @@
+
+
+The following code compares the two strings using a case insensitive ordinal comparison. Case-insensitive means "user" equals "User". The result is the same as comparing the value returned by <xref:System.String.ToUpperInvariant%2A?displayProperty=nameWithType> using an ordinal comparison. Run it to verify that the two strings are equal.
+
+[!code-csharp-interactive[Comparing strings ignoring case](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#2)]
+
+The <xref:System.String> class also contains static <xref:System.String.Equals%2A> methods you can use similar to the instance methods. The following code shows both case-sensitive and case-insensitive ordinal comparisons using the static methods: 
+
+[!code-csharp-interactive[Comparing for equality with static string methods](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#3)]
+
+The following example demonstrates the *string interning* feature of C#. When a program declares two or more identical string variables, the compiler stores them all in the same location. By calling the <xref:System.Object.ReferenceEquals%2A> method, you can see that the two strings actually refer to the same object in memory. Use the <xref:System.String.Copy%2A?displayProperty=nameWithType> method to avoid interning. After the copy has been made, the two strings have different storage locations, even though they have the same value. Run the following sample to show that strings `a` and `b` are *interned* meaning they share the same storage. The strings `a` and `c` are not.
+
+[!code-csharp-interactive[Demonstrating string interning](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#4)]
+
+## Ordinal sorting and searching strings
+
+The following examples show how to sort and search for strings in an array in a culture-sensitive manner by using the static <xref:System.Array> methods that take a <xref:System.StringComparer?displayProperty=nameWithType> parameter.
+
+This example shows how to search an array of strings using an ordinal search: 
+
+[!code-csharp-interactive[Sorting an array of strings](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#5)]
+
+Once the array is sorted, you can search for entries using a binary search. A binary search starts in the middle of the collection to determine which half of the collection would contain the sought string. Each subsequent comparison subdivides the remaining part of the collection in half. The following sample demonstrates the importance of using the same comparison rules for both sorting and searching. The array is sorted using <xref:System.StringComparer.Ordinal?displayProperty=nameWithType>. However, the search uses <xref:System.StringComparer.OrdinalIgnoreCase?displayProperty=nameWithType>. The local function `ShowWhere` displays information about where the string was found. If the string was not found, the returned value indicates where it would be if it were found.
```

Comment:
would contain --> contains
It's not clear without careful inspection (especially since the output display the search string, not the string found) how the example demonstrates the importance of using the same comparison rules -- either mention explicitly that what's found is a partly uppercase string that's not an exact match for an array element, or change the value displayed, as I mentioned below in the code.